AnkiDroid/src/main/java/com/ichi2/anki/StudyOptionsFragment.java

```diff
@@ -433,57 +435,56 @@ private void configureToolbarInternal(boolean recur) {
-                mLoadWithDeckOptions = false;
-                Deck deck = getCol().getDecks().current();
-                if (deck.isDyn() && deck.has("empty")) {
-                    deck.remove("empty");
-                }
-                    mProgressDialog = StyledProgressDialog.show(getActivity(), null,
-                            getResources().getString(R.string.rebuild_filtered_deck), true);
-                    TaskManager.launchCollectionTask(new CollectionTask.RebuildCram(), getCollectionTaskListener(true));
-            } else {
-                TaskManager.waitToFinish();
-                refreshInterface(true);
-            }
-        } else if (requestCode == AnkiActivity.REQUEST_REVIEW) {
-            if (resultCode == Reviewer.RESULT_NO_MORE_CARDS) {
-                // If no more cards getting returned while counts > 0 (due to learn ahead limit) then show a snackbar
-                if (getCol().getSched().count() > 0 && mStudyOptionsView != null) {
-                    View rootLayout = mStudyOptionsView.findViewById(R.id.studyoptions_main);
-                    UIUtils.showSnackbar(getActivity(), R.string.studyoptions_no_cards_due, false, 0, null, rootLayout);
-                }
+    });
+
+    ActivityResultLauncher<Intent> onDeckOptionsActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+        configureToolbar();
+        if (result.getResultCode() == DeckPicker.RESULT_DB_ERROR || result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED) {
+            closeStudyOptions(result.getResultCode());
+            return;
+        }
+        if (mLoadWithDeckOptions) {
+            mLoadWithDeckOptions = false;
+            Deck deck = getCol().getDecks().current();
+            if (deck.isDyn() && deck.has("empty")) {
+                deck.remove("empty");
             }
-        } else if (requestCode == STATISTICS && mCurrentContentView == CONTENT_CONGRATS) {
+            mProgressDialog = StyledProgressDialog.show(getActivity(), null,
+                    getResources().getString(R.string.rebuild_filtered_deck), true);
+            TaskManager.launchCollectionTask(new CollectionTask.RebuildCram(), getCollectionTaskListener(true));
+        } else {
+            TaskManager.waitToFinish();
+            refreshInterface(true);
+        }
+    });
+
+    ActivityResultLauncher<Intent> onStatisticsActivityResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), result -> {
+        configureToolbar();
+        if (result.getResultCode() == DeckPicker.RESULT_DB_ERROR || result.getResultCode() == DeckPicker.RESULT_MEDIA_EJECTED) {
+            closeStudyOptions(result.getResultCode());
+            return;
+        }
+        if (result.getData().hasExtra("originalDeck")) {
+            getCol().getDecks().select(result.getData().getLongExtra("originalDeck", 0L));
+        }
+        if(mCurrentContentView == CONTENT_CONGRATS) {
```

Comment:
trivial spacing change

```suggestion
        if (mCurrentContentView == CONTENT_CONGRATS) {
```